File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var i: usize= 0;
    var sum: usize = 0;

    while (i < number + 1) {
        sum += i;
        i += 1;
    }

    return sum * sum;
}

pub fn sumOfSquares(number: usize) usize {
    var i: usize = 0;
    var sum: usize = 0;

    while (i < number + 1) {
        sum += (i * i);
        i += 1;
    }

    return sum;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}

File: nim/difference-of-squares/difference_of_squares.nim
import math, sequtils

proc squareOfSum*(n: int): int = toSeq(1..n).sum ^ 2

proc sumOfSquares*(n: int): int = toSeq(1..n).mapIt( it ^ 2).sum

proc difference*(n: int): int = squareOfSum(n) - sumOfSquares(n)
